tests/decomp/020-cond-flags1/cond-flags1.lst.exp

```diff
@@ -3,11 +3,12 @@ void fun_05()
 {
   // $eax = $eax_0; // {'uses': ['10', '11']} (dead);
   // $ebx = $ebx_0; // {'uses': ['10', '11']} (dead);
-  $ecx = 0; // {'uses': ['40']}
-  // $Z = $eax_0 == $ebx_0; // {'uses': ['20']} (dead);
-  // $C = $eax_0 < $ebx_0; // {'uses': []} (dead);
   if ($eax_0 != $ebx_0) {
     $ecx = 1; // {'uses': ['40']}
+  } else {
+    $ecx = 0; // {'uses': ['40']}
+    // $Z = $eax_0 == $ebx_0; // {'uses': ['20']} (dead);
```

Comment:
Here's a smaller drawback, if you look at the internal representation. Dead statements get completely out of place with this code motion. Well, such is life, they are dead cruft, being pulled by the stronger forces. There're no dead statements in the final output, and someone looking into the intermediate representation know it's not piece of cake.

(Another note I'd have is that it morally prepares us to SSA. My big concern with it is that it may transform original code in quite unresembling itself ways. Well, the same happens with any advanced transform, as the experience shows. So again, I should be morally ready for SSA, now just someone needs to right converters into/out of SSA, LOL.)